src/main/scala/mesosphere/util/monitor/HeartbeatMonitor.scala

```diff
@@ -0,0 +1,81 @@
+package mesosphere.util.monitor
+
+import javax.inject.Inject
+
+import akka.actor._
+import akka.event.LoggingReceive
+import scala.concurrent.duration._
+
+// HeartbeatActor monitors the heartbeat of some process and executes handlers for various conditions.
+// If an expected heartbeat is missed then execute an `onSkipped` handler.
+// If X number of subsequent heartbeats are missed then execute an `onFailure` handler.
+// Upon creation the actor is in an inactive state and must be sent an EventActivate message to activate.
+// Once activated the actor will monitor for EventPulse messages (these are the heartbeats).
+// The actor may be deactivated by sending it an EventDeactivate message.
+class HeartbeatActor @Inject() (
+    config: HeartbeatActor.Config
+) extends Actor {
+
+  import HeartbeatActor._
+
+  def receive: Receive = stateInactive
+
+  def stateInactive: Receive = LoggingReceive.withLabel("inactive") {
```

Comment:
I actually found the akka FSM over complicated for this. But now that I've restructured things I'll take another look at it. I sure do miss the simplicity of other langs...